pkg/registry/ecr/registry.go

```diff
@@ -0,0 +1,136 @@
+package ecr
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"strings"
+
+	"github.com/v3io/registry-creds-handler/pkg/registry"
+	"github.com/v3io/registry-creds-handler/pkg/registry/abstract"
+	"github.com/v3io/registry-creds-handler/pkg/util"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/nuclio/errors"
+	"github.com/nuclio/logger"
+)
+
+type Registry struct {
+	*abstract.Registry
+	awsCreds registry.AWSCreds
+}
+
+func NewRegistry(parentLogger logger.Logger,
+	secretName string,
+	namespace string,
+	creds string,
+	registryUri string) (*Registry, error) {
+	newRegistry := &Registry{}
+
+	// create base
+	abstractRegistry, err := abstract.NewRegistry(parentLogger, newRegistry, secretName, namespace, creds, registryUri)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create abstract registry")
+	}
+
+	newRegistry.Registry = abstractRegistry
+	return newRegistry, nil
+}
+
+func (r *Registry) EnrichAndValidate() error {
+
+	if err := r.Enrich(); err != nil {
+		return errors.Wrap(err, "Failed to enrich ECR Registry")
+	}
+
+	if err := r.Validate(); err != nil {
+		return errors.Wrap(err, "Failed to validate ECR Registry")
+	}
+
+	return nil
+}
+
+func (r *Registry) Enrich() error {
```

Comment:
```suggestion
func (r *Registry) enrich() error {
```

no need to expose functions unless there is a good reason to